src/displayapp/DisplayApp.cpp

```diff
@@ -505,6 +505,7 @@ void DisplayApp::LoadNewScreen(Apps app, DisplayApp::FullRefreshDirections direc
   // This is mainly to fix an issue with receiving two notifications at the same time
   // and shouldn't happen otherwise.
   if (app != currentApp) {
+    popupMessage.SetHidden(true);
```

Comment:
please add an additional comment describing why we do SetHidden here, otherwise the comment above doesn't fully  represent the code below anymore. Other than that big THANKS for the fix!